File: Python/Math-Array-String/15.py
# 709. To Lower Case


# class Solution:
#     def toLowerCase(self, s: str) -> str:
#         return s.lower()


class Solution:
    def toLowerCase(self, s: str) -> str:
        lower_str = ""
        for char in s:
            ascii_value = ord(char)
            if 65 <= ascii_value <= 90:
                lower_str = lower_str + chr(ascii_value + 32)
            else:
                lower_str = lower_str + char
        return lower_str


sol = Solution()
output = sol.toLowerCase("LOVELY")
print(output)

File: JavaScript/Math-Array-String/15.js
// 709. To Lower Case

/**
 * @param {string} s
 * @return {string}
 */
var toLowerCase = function(s) {
    let result = "";
    for (const char of s) {
        const asciiIndex = char.charCodeAt(0);

        if (asciiIndex >= 65 && asciiIndex <= 90) {
            result += String.fromCharCode(asciiIndex + 32);
        }
        else {
            result += char;
        }
    }
    return result;
};

// var toLowerCase = function(s) {
//     return s.toLowerCase()
// };

console.log(toLowerCase("Hello World")); // Output: "hello world"